### PR TITLE
Implement internal TLS

### PR DESCRIPTION
**Pull request by fabricetriboix**
Internal TLS implemented for a dummy service I created, called `test-backend`. This will serve as a blueprint for real services, such as Alfresco, Pentaho and ArkCase.

### CODE REVIEW

File: Docker/envoy-tls/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+version: "2.4"  # docker-compose version 3 doesn't support "init: true"
+
+services:
+  test-backend:
+    depends_on: [ envoy-tls ]
+    image: 300674751221.dkr.ecr.us-west-1.amazonaws.com/test-backend:2
```
Comment (david-oc-miller) [inline Docker/envoy-tls/docker-compose.yml:6]: I guess this ECR reference will be parameterized eventually, or is this only a sample project?  If somebody runs this code in their own account they would change this reference, right?

File: Docker/envoy-tls/envoy-tls/envoy-config.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,51 @@
+static_resources:
+  listeners:
+    - address:
+        socket_address: { address: 0.0.0.0, port_value: 8443 }
+      filter_chains:
+        transport_socket:
+          name: envoy.transport_sockets.tls
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext
```
Comment (david-oc-miller) [inline Docker/envoy-tls/envoy-tls/envoy-config.yaml.tmpl:9]: Which TLS protocol versions are enabled?  Can we restrict it to only TLSv1.2, or even TLSv1.3?

File: Docker/envoy-tls/docker-compose.yml (hunk 1 above)
Comment (fabricetriboix) [reply Docker/envoy-tls/docker-compose.yml:6]: @david-oc-miller Actually, I just did that right at the same time you wrote that comment!

Comment (fabricetriboix) [reply Docker/envoy-tls/docker-compose.yml:6]: @david-oc-miller It's kind of a reference project to build, test and push the Docker images. It's quite difficult to parameterize things in docker-compose, unfortunately...

File: Docker/envoy-tls/envoy-tls/envoy-config.yaml.tmpl (hunk 2 above)
Comment (fabricetriboix) [reply Docker/envoy-tls/envoy-tls/envoy-config.yaml.tmpl:9]: @david-oc-miller Yes, that's a good point. I will have a look at how to restrict that.

File: Docker/envoy-tls/docker-compose.yml (hunk 1 above)
Comment (david-oc-miller) [reply Docker/envoy-tls/docker-compose.yml:6]: Can we lift it up into an environment variable, in some more obvious part of the docker-compose?

Comment (fabricetriboix) [reply Docker/envoy-tls/docker-compose.yml:6]: @david-oc-miller Yes, we can. However, I am not sure whether it is a good idea or not. In my experience, I never remember to set the environment variables before running docker-compose... An alternative would be to wrap the invocation of docker-compose in a shell script, which would be much more user-friendly and safer in my opinion. Please let me know what you think.

Comment (david-oc-miller) [reply Docker/envoy-tls/docker-compose.yml:6]: Wrapping in a shell script sounds great, thanks.

Comment (fabricetriboix) [reply Docker/envoy-tls/docker-compose.yml:6]: @david-oc-miller All right, will do!

**Pull request merged**